jerry-core/config.h

```diff
@@ -42,15 +42,20 @@
  * Size of heap
  */
 #ifndef CONFIG_MEM_HEAP_AREA_SIZE
-# define CONFIG_MEM_HEAP_AREA_SIZE (256 * 1024)
-#elif CONFIG_MEM_HEAP_AREA_SIZE > (256 * 1024)
-# error "Currently, maximum 256 kilobytes heap size is supported"
+# define CONFIG_MEM_HEAP_AREA_SIZE (512 * 1024)
+#elif CONFIG_MEM_HEAP_AREA_SIZE > (512 * 1024)
+# error "Currently, maximum 512 kilobytes heap size is supported"
 #endif /* !CONFIG_MEM_HEAP_AREA_SIZE */
 
 /**
+ * Max heap usage limit
+ */
+#define CONFIG_MEM_HEAP_MAX_LIMIT 8192
```

Comment:
It defines a maximum value for the heap limit. Heap limit is, as the name suggest, a limit of memory usage, and whenever the size of allocated memory since the previous limit would hit the next one, we would run garbage collection. The limit was previously set to `HEAP_AREA_SIZE / 32 == 8192`. Because I increased the size of the heap, garbage collection became less frequent, so I added a maximum value. This way it wouldn't affect environments where the heap size is different.